auth_ldap_user_pattern/models/res_company_ldap.py

```diff
@@ -0,0 +1,41 @@
+# Copyright (C) 2019 IBM Corp.
+# Copyright (C) 2019 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class CompanyLDAP(models.Model):
+    _inherit = "res.company.ldap"
+
+    user_pattern = fields.Char(
+        string='User Pattern',
+        size=128
+    )
+
+    def map_ldap_attributes(self, conf, login, ldap_entry):
+        res = super(CompanyLDAP, self).map_ldap_attributes(
+            conf=conf, login=login, ldap_entry=ldap_entry)
+        res.update({'is_ldap_user': True})
+        return res
+
+    @api.multi
+    def get_ldap_dicts(self):
+        res = super(CompanyLDAP, self).get_ldap_dicts()
```

Comment:
please explain this override

Reply:
@hbrunn : Purpose to read "user_pattern" field. Now method is optimized in recent commit.

Reply:
I see, thanks